### DISCUSSION TITLE
Version with new calender view v24.1.7-Database

### DISCUSSION THREAD

**Discussion by MartinNK** [RaBa64/OpenDTU#20] [General]
@RaBa64 Great ... but some problems with display on website ...

![Screenshot_20240109_144359](https://github.com/RaBa64/OpenDTU-Database/assets/3524916/025f4fca-2388-40a7-a23a-0437b729f9b0)

- Legend does not scale on bigscreens and last month short name is cropped at the right side ... 
- missing display of year and may it will be better to show lines between monthes and years (maybe a little bit thiker)
- maybe only show dots in calender with real date.

**Comment by RaBa64** [comment]
I fixed the first point. I will publish a new release later this evening.

The new calendar view always shows the last 365 days (rounded up to full weeks), so I don't think it is necessary to waste space and display the year (if in doubt, hover over a little square or look on the calendar on the wall ;-) )

I don't get the last point.
All little squares are real dates. Dates without data are shown in grey, dates with data are shown in blue.
When you hover over a little square, you will se the date and Wh produced.

What is still open is to switch to dark mode. I hope I can also fix this today.

**Reply by MartinNK** [reply]
Hast Du mal gecheckt wieviele Tage Du im Filesystem unterbringst? Reicht ja wohl etwa für 1-2 Jahre nach meiner Schätzung ;-)

**Reply by kinichahau2023** [reply]
Stufe 1 (von den 5 möglichen) grafischen Rundungen von den (Ertrags-)Würfeln im Kalender wäre auch schick, da die übrigen "Würfel" im Hauptprogramm auch geringfügig rundlich sind.

**Reply by RaBa64** [reply]
> Hast Du mal gecheckt wieviele Tage Du im Filesystem unterbringst? Reicht ja wohl etwa für 1-2 Jahre nach meiner Schätzung ;-)
> 
> 

Ich habe alles seit Mai 2023 am laufen und habe in diesen knapp 9 Monaten 52KB von 140KB verbraucht. Davon waren aber schon etliche KB belegt, auch ohne Datenbank. Ich schätze den Speicherbedarf auf ca. 4KB pro Monat, also sollte der restliche Speicher noch für ca. 22 Monate ausreichen. In Summe sollten also ca. 31 Monate oder  2,5 Jahre hineinpassen. Es sollte aber auch Luft da sein dass Filesystem zu vergrößern.

**Comment by MartinNK** [comment]
Hallo Ralf,

danke für die Infos ... hab schon den Eintrag gefunden von Dir auf Github ber 
Speicherbedarf von vor einiger Zeit.

Leider habe ich bei der Umstellung meine Datenbank intern verloren und 
irgendetwas hakte nach dem Update über Webinterface. Kann es leider nicht mehr 
nachvollziehen.

Hab das konmplett neu gefläscht ;-) und leider wegen der fehlenden Funktion zum 
zurückspielen der Database nun noch den Stand seit 4.Januar drauf.

Wie siehst Du die Möglichkeit ein implementieren des Wiederherstellen der DB 
Datei ... leider bin ich noch einen Generation ohne + oder ++ Sprachen, leider 
nicht mehr mein Ding. Nur ein bissle JS schaffe ich aber doch noch :-)

Hab mal nach verschiedenen Alternativen für Google Charts gesucht ... bin 
dabei auf den folgenden Artikel gestoßen ... https://www.slant.co/options/
11578/alternatives/~google-charts-alternatives

Gruß 
Martin


Am Samstag, 20. Januar 2024, 18:43:26 CET schrieb Ralf Bauer:
> > Hast Du mal gecheckt wieviele Tage Du im Filesystem unterbringst? Reicht
> > ja wohl etwa für 1-2 Jahre nach meiner Schätzung ;-)
> Ich habe alles seit Mai 2023 am laufen und habe in diesen knapp 9 Monaten
> 52KB von 140KB verbraucht. Davon waren aber schon etliche KB belegt, auch
> ohne Datenbank. Ich schätze den Speicherbedarf auf ca. 4KB pro Monat, also
> sollte der restliche Speicher noch für ca. 22 Monate ausreichen. In Summe
> sollten also ca. 31 Monate oder  2,5 Jahre hineinpassen. Es sollte aber
> auch Luft da sein dass Filesystem zu vergrößern.

**Comment by MartinNK** [comment]
Hallo Ralf,danke für die Infos ... hab schon den Eintrag gefunden von Dir auf Github ber Speicherbedarf von vor einiger Zeit.Leider habe ich bei der Umstellung meine Datenbank intern verloren und irgendetwas hakte nach dem Update über Webinterface. Kann es leider nicht mehr nachvollziehen.Hab das konmplett neu gefläscht ;-) und leider wegen der fehlenden Funktion zum zurückspielen der Database nun noch den Stand seit 4.Januar drauf.Wie siehst Du die Möglichkeit ein implementieren des Wiederherstellen der DB Datei ... leider bin ich noch einen Generation ohne + oder ++ Sprachen, leider nicht mehr mein Ding. Nur ein bissle JS schaffe ich aber doch noch :-)Hab mal nach verschiedenen Alternativen für Google Charts gesucht ... bin dabei auf den folgenden Artikel gestoßen ... https://www.slant.co/options/11578/alternatives/~google-charts-alternativesGruß MartinAm Samstag, 20. Januar 2024, 18:43:26 CET schrieb Ralf Bauer:> > Hast Du mal gecheckt wieviele Tage Du im Filesystem unterbringst? Reicht> > ja wohl etwa für 1-2 Jahre nach meiner Schätzung ;-)> Ich habe alles seit Mai 2023 am laufen und habe in diesen knapp 9 Monaten> 52KB von 140KB verbraucht. Davon waren aber schon etliche KB belegt, auch> ohne Datenbank. Ich schätze den Speicherbedarf auf ca. 4KB pro Monat, also> sollte der restliche Speicher noch für ca. 22 Monate ausreichen. In Summe> sollten also ca. 31 Monate oder  2,5 Jahre hineinpassen. Es sollte aber> auch Luft da sein dass Filesystem zu vergrößern.Von meinem/meiner Galaxy gesendet

**Comment by MartinNK** [comment]
Dear Ralf ( @RaBa64 ),

any development? Did you find an alternative for Google charts?
Is it possible to have a new version with newest original openDTU?

Thanks a lot for your work
Martin

**Comment by MartinNK** [comment]
It will be sad if that great forge will stop being developed ;-(

**Reply by kinichahau2023** [reply]
Schließe mich uneingeschränkt an ... ;-(
